### PR TITLE
BSIDM-352: So Cancel credit mngm

### PR DESCRIPTION
**Pull request by leemannd**
[ADD]Give back credit to partner when the SO is canceled

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
HISTORY?!

**Comment by leemannd** [comment]
@oleksandrpaziuk There is no History in this repo
version number is updated in https://github.com/camptocamp/odoo-sale-addons/pull/7/files#diff-a5b2a165ed22246705af9d75c15c33a3R6

**Review by ghost — APPROVED** [review]
LGTM! :+1:

File: sale_credit_point/models/sale_order.py [hunk 1]
```diff
@@ -34,3 +34,13 @@ def action_confirm(self):
...
         return super().action_confirm()
+
+    @api.multi
+    def action_cancel(self):
+        for sale in self:
+            if sale.state == 'sale':
+                sale.partner_id.credit_point_increase(
+                    sale.amount_total, "Sale Order canceled")
+                sale.partner_id.update_history(
+                    sale.amount_total, "increase", "Sale Order canceled")
```
Comment (guewen) [inline sale_credit_point/models/sale_order.py]: No translations?

**Review by guewen — CHANGES_REQUESTED** [review]

File: sale_credit_point/models/sale_order.py (hunk 1 above)
Comment (guewen) [reply sale_credit_point/models/sale_order.py]: Is it normal that you have 2 calls which seems pretty similar (disclaimer: I didn't looked at the methods)

**Comment by leemannd** [comment]
@guewen Yep, and tha't who it is used for instance into the wizard

**Review by simahawk — APPROVED** [review]
quick eyeball review: LG

**Review by guewen — APPROVED** [review]

**Comment by simahawk** [comment]
@leemannd you'll have to rebase and raise version again after https://github.com/camptocamp/odoo-sale-addons/pull/7 is merged ;)

**Comment by leemannd** [comment]
Version number raised

**Comment by leemannd** [comment]
Fast tracking. Was green before rebase/squash

**Pull request merged**